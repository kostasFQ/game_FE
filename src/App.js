import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { Switch, Route, BrowserRouter as Router } from "react-router-dom";
// import TopMenu from 'components/TopMenu/container';
import NotFoundPage from 'views/NotFoundPage';
import GamePage from 'views/GamePage/container';
import HomePage from 'views/HomePage';
import LoginPage from 'views/LoginPage';
import Timer from 'components/Timer/container';


function App(props) {
  return (
    <Fragment>
      <Router>
        <Timer />
      {/* <TopMenu /> */}
        <Switch>
          <Route path='/' exact component={HomePage} />
          <Route path='/login' component={LoginPage} />
          <Route path='/game' component={GamePage} />
          <Route component={NotFoundPage} />
        </Switch>
      </Router>
    </Fragment>
  );
}

const mapStateToProps = state => ({
  state: state.user,
});

export default connect(mapStateToProps)(App);
