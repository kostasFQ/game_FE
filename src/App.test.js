import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { StaticRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from './App';
import _store from '../mocks/store';

const initialState = { game: _store };
const mStore = configureMockStore([thunk]);
const store = mStore(initialState);

const instance = renderer.create(
  <Provider store={store}>
    <StaticRouter location="/" context={{}}>
      <App />
    </StaticRouter>
  </Provider>,
);

test('FinalCount should be render', () => {
  let component = instance.toJSON();
  expect(component).toMatchSnapshot();
})
