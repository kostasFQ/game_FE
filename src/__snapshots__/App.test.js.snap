// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinalCount should be render 1`] = `
Array [
  <div
    className="title__container"
  >
    <div
      className="title__container__text"
    >
      clickatron3000
    </div>
  </div>,
  <div
    className="timersContainer"
  >
    <div
      className="timersContainer__counter_hide"
    >
      <span
        className="timersContainer__digits"
      />
       seconds remain
    </div>
  </div>,
  <div
    className="homePage_container"
  >
    <div>
      select how many seconds you want
    </div>
    <form
      className="setTimers__container"
      onClick={[Function]}
    >
      <div
        className="setTimers__container__checkboxes"
      >
        <input
          className="setTimers__container__checkboxes__item"
          disabled={false}
          name="seconds"
          type="button"
          value={5}
        />
        <input
          className="setTimers__container__checkboxes__item"
          disabled={false}
          name="seconds"
          type="button"
          value={10}
        />
        <input
          className="setTimers__container__checkboxes__item setTimers__container__checkboxes__item_active"
          disabled={true}
          name="seconds"
          type="button"
          value={15}
        />
        <input
          className="setTimers__container__checkboxes__item"
          disabled={false}
          name="seconds"
          type="button"
          value={30}
        />
        <input
          className="setTimers__container__checkboxes__item"
          disabled={false}
          name="seconds"
          type="button"
          value={60}
        />
      </div>
    </form>
    <button
      className="startButton"
      disabled={false}
      onClick={[Function]}
    >
      START!
    </button>
    <h3>
      Loading...
    </h3>
  </div>,
]
`;
